File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.py
from typing import List
from collections import Counter

class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        n = len(nums)

        count = Counter(nums)

        # Every element forms its own subarray
        if k == 1:
            ans = -1

            for x in nums:
                if count[x] == 1:
                    ans = max(ans, x)

            return ans

        # Entire array is one subarray
        if k == n:
            return max(nums)

        # Only first and last elements can qualify
        ans = -1

        if count[nums[0]] == 1:
            ans = max(ans, nums[0])

        if count[nums[-1]] == 1:
            ans = max(ans, nums[-1])

        return ans

File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.java
class Solution {
    public int largestInteger(int[] nums, int k) {
        int n = nums.length;

        int[] count = new int[51];

        for (int x : nums) {
            count[x]++;
        }

        // Every element is its own subarray
        if (k == 1) {
            int ans = -1;

            for (int x : nums) {
                if (count[x] == 1) {
                    ans = Math.max(ans, x);
                }
            }

            return ans;
        }

        // Entire array is one subarray
        if (k == n) {
            int ans = 0;

            for (int x : nums) {
                ans = Math.max(ans, x);
            }

            return ans;
        }

        // Only first and last elements can qualify
        int ans = -1;

        if (count[nums[0]] == 1) {
            ans = Math.max(ans, nums[0]);
        }

        if (count[nums[n - 1]] == 1) {
            ans = Math.max(ans, nums[n - 1]);
        }

        return ans;
    }
}